src/components/scenes/EdgeLoginScene.tsx

```diff
@@ -1,242 +1,118 @@
-import { EdgeLobby } from 'edge-core-js'
 import * as React from 'react'
-import { ActivityIndicator, StyleSheet, Text, View } from 'react-native'
-import FastImage from 'react-native-fast-image'
-import { isIPhoneX } from 'react-native-safe-area-view'
+import { ActivityIndicator, Image, View } from 'react-native'
 import { sprintf } from 'sprintf-js'
 
 import { lobbyLogin } from '../../actions/EdgeLoginActions'
+import { useAsyncEffect } from '../../hooks/useAsyncEffect'
 import s from '../../locales/strings'
-import { PrimaryButton } from '../../modules/UI/components/Buttons/PrimaryButton.ui'
-import { SecondaryButton } from '../../modules/UI/components/Buttons/SecondaryButton.ui'
 import { config } from '../../theme/appConfig'
-import { THEME } from '../../theme/variables/airbitz'
-import { connect } from '../../types/reactRedux'
-import { NavigationBase, NavigationProp } from '../../types/routerTypes'
+import { useDispatch, useSelector } from '../../types/reactRedux'
+import { NavigationBase, RouteProp } from '../../types/routerTypes'
+import { WarningCard } from '../cards/WarningCard'
 import { SceneWrapper } from '../common/SceneWrapper'
-
-interface OwnProps {
-  navigation: NavigationProp<'edgeLogin'>
-}
-interface StateProps {
-  error: string | null
-  isProcessing: boolean
-  lobby: EdgeLobby | null
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+import { TitleText } from '../text/TitleText'
+import { Fade } from '../themed/Fade'
+import { MainButton } from '../themed/MainButton'
+import { LoadingScene } from './LoadingScene'
+interface Props {
+  navigation: NavigationBase
+  route: RouteProp<'edgeLogin'>
 }
 
-interface DispatchProps {
-  accept: (navigation: NavigationBase) => void
-}
+export const EdgeLoginScene = (props: Props) => {
+  const { navigation, route } = props
+  const { account } = useSelector(state => state.core)
+  const { loginToken } = route.params
+  const [isLobby, setIsLobby] = React.useState<boolean>(false)
+  const [isApproved, setIsApproved] = React.useState<boolean>(false)
+  const isLoading = React.useMemo(() => {
+    return !isLobby || isApproved
+  }, [isLobby, isApproved])
```

Comment:
We don't actually need a `useMemo` on this, since it's simple boolean (so `===` works) and it's cheap to calculate. Just do `const isLoading = !isLobby || isApproved`, and you'll have the exact same result.

Well, actually, with the other refactors, we will probably also need `const isLobby = lobby != null`, but yeah, same thimg.